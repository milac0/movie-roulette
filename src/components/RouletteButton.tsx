import React from "react";
import { css } from "emotion";
import { colors } from "../theme/theme";
import shuffleImg from "../assets/images/shuffle2.png";
import Modal from "@material-ui/core/Modal";

const rouletteBtn = css`
  background: ${colors.secondary};
  border-radius: 50%;
  bottom: 1.75em;
  height: 60px;
  position: fixed;
  right: 4em;
  width: 60px;
  z-index: 1;
`;

const modalContent = css`
  background: ${colors.backgroundGrey};
  border: "2px solid #000";
  left: 50%;
  margin: 0;
  padding: 1em;
  position: absolute;
  top: 50%;
  transform: translate(-50%, -50%);
`;

interface Props {}

const RouletteButton: React.FC<Props> = () => {
  const [open, setOpen] = React.useState(false);
  const handleShuffle = () => {
    setOpen(true);
  };
  return (
    <div>
      <img src={shuffleImg} className={rouletteBtn} onClick={handleShuffle} />
      <Modal open={open} onClose={() => setOpen(false)}>
        <div className={modalContent}>
          <p>modal</p>
        </div>
      </Modal>
    </div>
  );
};

export default RouletteButton;
