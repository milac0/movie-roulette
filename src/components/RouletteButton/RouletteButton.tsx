import React from "react";
import shuffleImg from "../../assets/images/shuffle.png";
import Modal from "@material-ui/core/Modal";
import ModalContent from "../ModalContent/ModalContent";
import styles from "./rouletteButton.scss";

interface Props {}

const RouletteButton: React.FC<Props> = () => {
  const [open, setOpen] = React.useState(false);

  const handleOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  return (
    <div>
      <img
        src={shuffleImg}
        className={styles.rouletteBtn}
        onClick={handleOpen}
      />
      <Modal open={open} onClose={handleClose}>
        <div>
          <ModalContent handleClose={handleClose} />
        </div>
      </Modal>
    </div>
  );
};

export default RouletteButton;
