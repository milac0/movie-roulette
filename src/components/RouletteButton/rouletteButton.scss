@import "../../theme/variables";

.rouletteBtn {
  background: $secondary;
  border: 3px solid $primary;
  border-radius: 50%;
  bottom: 3em;
  height: 60px;
  position: fixed;
  right: 3em;
  width: 60px;
  z-index: 1;

  &:hover {
    background: $secondaryDark;
    cursor: pointer;
  }
}
