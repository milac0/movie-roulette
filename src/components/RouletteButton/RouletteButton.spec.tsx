import React from 'react';
import renderer from 'react-test-renderer';

import RouletteButton from './RouletteButton';

describe('RouletteButton renders correctly', () => {
    test('snapshot renders', () => {
        const tree = renderer.create(<RouletteButton />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

