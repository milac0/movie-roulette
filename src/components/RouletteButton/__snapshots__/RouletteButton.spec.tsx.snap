// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouletteButton renders correctly snapshot renders 1`] = `
<div>
  <img
    className="rouletteBtn"
    onClick={[Function]}
    src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNk+P+/HgAFhAJ/wlseKgAAAABJRU5ErkJggg=="
  />
</div>
`;
